app/src/main/java/com/hedvig/app/feature/marketpicker/MarketPickerActivity.kt

```diff
@@ -0,0 +1,117 @@
+package com.hedvig.app.feature.marketpicker
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.localbroadcastmanager.content.LocalBroadcastManager
+import androidx.preference.PreferenceManager
+import androidx.recyclerview.widget.DividerItemDecoration
+import com.hedvig.app.BaseActivity
+import com.hedvig.app.R
+import com.hedvig.app.feature.language.LanguageSelectionTracker
+import com.hedvig.app.feature.language.LanguageViewModel
+import com.hedvig.app.feature.marketing.ui.MarketingActivity
+import com.hedvig.app.feature.settings.Language
+import com.hedvig.app.feature.settings.SettingsActivity
+import com.hedvig.app.makeLocaleString
+import com.hedvig.app.util.extensions.compatDrawable
+import com.hedvig.app.util.extensions.getStoredBoolean
+import com.hedvig.app.util.extensions.observe
+import com.hedvig.app.util.extensions.storeBoolean
+import com.hedvig.app.util.extensions.view.setHapticClickListener
+import kotlinx.android.synthetic.main.activity_market_picker.*
+import org.koin.android.ext.android.inject
+import org.koin.android.viewmodel.ext.android.viewModel
+
+class MarketPickerActivity : BaseActivity(R.layout.activity_market_picker) {
+    private val marketModel: MarketPickerViewModel by viewModel()
+    private val languageViewModel: LanguageViewModel by viewModel()
+    private val tracker: LanguageSelectionTracker by inject()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        storeBoolean(HAS_SHOWN_MARKET_SELECTION, true)
+
+        var marketAdapter: MarketAdapter? = null
```

Comment:
Can we initialize the MarketAdapter with an empty list and update the data later on?

Also, should we hide these elements while loading the Geolocation?